docs/projects/reverse-a-string/_posts/2018-12-24-python.md

```diff
@@ -0,0 +1,126 @@
+According to Python documentation, `argv` is the list of command line arguments.
+
+With that in mind, it’s clear that we’re just verifying its length. But, why
+aren’t we just checking to see if the list is empty? That would probably be
+more efficient and more explicit. Well, as it turns out, the interpreter stores
+the script name as the first argument, so `argv` will always contain at least one
+element.
+
+If we put all that information together, we’ll notice that we only branch if the
+list of command line arguments contains more than one item. If it does, we know
+the user has passed us some string.[^1]
+
+### The Slice & Print
+
+Our last line is a bit terse, but we can handle it:
+
+```python
+print(sys.argv[1][::-1])
+```
+
+The first thing to note is the print statement. We’ll probably remember that
+from our Hello World experience, so no need to dig into that much deeper.
+
+Inside the print statement, we have quite a bit of syntax. First, we access our
+command line arguments again using sys.argv. However, this time we’re trying to
+access one of the items using an index of one. Like many languages, Python list
+indices start from zero, so clearly we’re trying to access our string for reversal.
+
+After that, we’ll notice another piece of interesting syntax: [::-1]. It looks
+sort of like list access, but it has colons and a negative value. As it turns
+out, that syntax is called a slice. Typically, slices are used to get a subset
+of an iterable. For instance, if I wanted the first four characters of a string,
+I could do the following:
+
+```python
+myString = "Hello, World!"
+print(myString[:4])  # Prints "Hell"
+```
+
+Okay, maybe that’s not a family friendly example, but you get the idea.
+
+Now, with a single colon, we can slice between any two indices. With a second
+colon, we can declare the step. For example, maybe we want every other character
+in a string:
+
+```python
+myString = "Hello, World!"
+print(myString[::2])  # Prints "Hlo ol!"
+```
+
+As an added wrinkle, we can step backwards through the string using a negative
+value. That’s exactly how we reverse the string in our example.
+
+So, let’s put it all together. We get our string off the command line, take a
+slice of it backwards, and print the result. How cool is that?[^1]
+
+## How to Run Solution
+
+To run the Reverse a String in Python program, grab a copy of the Python file
+from GitHub. After that, get the latest version of Python. Now, all you have to
+do is run the following from the command line:
+
+```console
+python reverse-string.py "Hello, World!"
+```
+
+Alternatively, you can always copy the source code into an online Python
+interpreter. Just make sure you pass some input to your script before you run
+it.[^1]
+
+---
+
+#### References
+
+[^1]: J. Grifski, “Reverse a String in Python,” The Renegade Coder, 21-Apr-2018. [Online]. Available: <https://therenegadecoder.com/code/reverse-a-string-in-java/>. [Accessed: 22-Dec-2018].
```

Comment:
The reference links to the Java version of the article.
```suggestion
[^1]: J. Grifski, “Reverse a String in Python,” The Renegade Coder, 21-Apr-2018. [Online]. Available: <https://therenegadecoder.com/code/reverse-a-string-in-python/>. [Accessed: 22-Dec-2018].
``` 
should fix it :)

Reply:
Nice catch!